nixos/tests/all-tests.nix

```diff
@@ -291,6 +291,7 @@ in
   syncthing-relay = handleTest ./syncthing-relay.nix {};
   systemd = handleTest ./systemd.nix {};
   systemd-analyze = handleTest ./systemd-analyze.nix {};
+  systemd-boot = handleTestOn ["x86_64-linux"] ./systemd-boot.nix {};
```

Comment:
I don't think this is x86_64-linux specific. systemd-boot should work on aarch64, too.

Reply:
Indeed, it would be nice to extend this to aarch64. Currently I think `qemu-vm.nix` only works with x86_64. We'd need to use `AAVMF` instead of `OVMF.fd`, among possibly other things. I think this should be in another PR, however.

Reply:
I think it does work - at least we build `nixosTests.installer.simpleUefiSystemdBoot` for aarch64, too: https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.installer.simpleUefiSystemdBoot.aarch64-linux

Reply:
So the installer test `simpleUefiSystemdBoot` actually has some special logic for aarch64 which does not currently have an analogue in `qemu-vm.nix`:
https://github.com/NixOS/nixpkgs/blob/694ac1b127ce34fb3bf5890232e1375b3911867c/nixos/tests/installer.nix#L72

It doesn't look very complicated, but didn't immediately work when I tried moving that logic into `qemu-vm.nix`.  I'll keep working on it, but I think it'd be better to put it in a new PR where we try to get `nixos-rebuild build-vm` with UEFI working on aarch64.

Reply:
Fine to touch this in a future PR :-)